### ISSUE TITLE
result contain some characters that do not in my unicharset

### ISSUE THREAD

**Issue by yixinuestc** [tesseract-ocr/tesseract#2358]
### Environment

* **Tesseract Version**:  Tesseract v4.0.0
* **Commit Number**: b49806766eb189f0e0dd63f8592c542fe0f37ab6
* **Platform**: uBuntu 16.04 64bits
### Current Behavior:

My unicharset contains some English letters and some Chinese characters, the total number of characters is no more than 100. All the English letters are match with my unicharset,but Chinese characters are not match.

I tried to train in the following ways:

Fine tuning
 src/training/tesstrain.sh --fonts_dir /usr/share/fonts --training_text ../training_data/chi_sim_tuned.txt \
--langdata_dir ../langdata_lstm --tessdata_dir ./tessdata --lang chi_sim --linedata_only --noextract_font_properties  --exposures "0" \
--workspace_dir ~/share/workspace/tmp \
--save_box_tiff \
 --fontlist  "NSimSun" \
        "Times New Roman" \
       "Arial Unicode MS" \
       "SimSun" \
      "Merchant Copy" \
      "Merchant Copy Doublesize" \
       "Noto Sans CJK SC" \
"Noto Sans Mono CJK SC" \
--output_dir ~/tesstutorial/chi_sim_train \
--overwrite

mkdir -p ~/tesstutorial/chi_sim_tuned_from_chi_sim

combine_tessdata -e ../tessdata_best/chi_sim.traineddata ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim.lstm

lstmtraining --model_output ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim_tuned \
--continue_from ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim.lstm \
--traineddata ~/tesstutorial/chi_sim_train/chi_sim/chi_sim.traineddata \
--old_traineddata ../tessdata_best/chi_sim.traineddata \
--train_listfile ~/tesstutorial/chi_sim_train/chi_sim.training_files.txt \
--max_iterations 3000

lstmtraining --stop_training --continue_from ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim_tuned_checkpoint  \
           --traineddata ~/tesstutorial/chi_sim_train/chi_sim/chi_sim.traineddata --model_output ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim_tuned.traineddata

Using this chi_sim_tuned.traineddata, the result contain some characters that do not in mycharset.

I also use fine tuning this way:

lstmtraining --model_output ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim_layer  \
--continue_from ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim.lstm \
--traineddata ~/tesstutorial/chi_sim_train/chi_sim/chi_sim.traineddata \
--append_index 5 --net_spec '[Lfx192 O1c1]' \
--train_listfile ~/tesstutorial/chi_sim_train/chi_sim.training_files.txt \
--max_iterations 3600

The result is same.

I even tried to train from scratch
lstmtraining --debug_interval 0 \
--model_output ~/tesstutorial/chi_sim_scratch/chi_sim \
--traineddata ~/tesstutorial/chi_sim_train/chi_sim/chi_sim.traineddata \
--net_spec '[1,48,0,1 Ct3,3,16 Mp3,3 Lfys64 Lfx96 Lrx96 Lfx256 O1c1]' \
 --learning_rate 20e-4 \
--train_listfile ~/tesstutorial/chi_sim_train/chi_sim.training_files.txt \
--max_iterations 400 

lstmtraining --stop_training --continue_from ~/tesstutorial/chi_sim_scratch/chi_sim_checkpoint  \
           --traineddata ~/tesstutorial/chi_sim_train/chi_sim/chi_sim.traineddata --model_output ~/tesstutorial/chi_sim_scratch/chi_sim_scratch.traineddata

Also the result contains some characters that do not in my unicharset.
### Expected Behavior:

All the characters in the result file should match the unicharset.

**Comment by Shreeshrii** [comment]
```
I also use fine tuning this way:

lstmtraining --model_output ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim_layer 
--continue_from ~/tesstutorial/chi_sim_tuned_from_chi_sim/chi_sim.lstm 
--traineddata ~/tesstutorial/chi_sim_train/chi_sim/chi_sim.traineddata 
--append_index 5 --net_spec '[Lfx192 O1c1]' 
--train_listfile ~/tesstutorial/chi_sim_train/chi_sim.training_files.txt 
--max_iterations 3600

The result is same.
```

Replacing a layer training needs larger number of iterations.

```
I even tried to train from scratch
lstmtraining --debug_interval 0 
--model_output ~/tesstutorial/chi_sim_scratch/chi_sim 
--traineddata ~/tesstutorial/chi_sim_train/chi_sim/chi_sim.traineddata 
--net_spec '[1,48,0,1 Ct3,3,16 Mp3,3 Lfys64 Lfx96 Lrx96 Lfx256 O1c1]' 
--learning_rate 20e-4 
--train_listfile ~/tesstutorial/chi_sim_train/chi_sim.training_files.txt 
--max_iterations 400
```

400 is very little for training from scratch. It should be high enough to get an error rate of 0.x.

**Comment by Shreeshrii** [comment]
>the result contains some characters that do not in my unicharset.

Please make a list of the characters in error.

**Comment by yixinuestc** [comment]
Yes,my iteration number is very small,but firstly, I am not care the accuracy of the result,I now just want to make sure all the result characters match my unicharset. Even the iteration number is very small, they should be match,right?
I attach my train_text file and unicharset in the attachment.
[attchment.zip](https://github.com/tesseract-ocr/tesseract/files/3016939/attchment.zip)

And the result contains some characters like "二" ，“六”，“让”，“询”，“【”   etc.
They are not in the train_text file.

**Comment by Shreeshrii** [comment]
@bertsky Do you know of any reason why the result would contain characters outside the unicharset for `chi_sim` - anything to do with recoder and `https://github.com/tesseract-ocr/langdata_lstm/blob/master/radical-stroke.txt` used for compressing CJK languages?

**Comment by Shreeshrii** [comment]
Possible reasons ...

chi_sim.config file loads `chi_sim_vert` - the characters could be from that.

You can remove/comment out first line from the config file and test.

**Comment by Shreeshrii** [comment]
Please test with traineddata in attached zip file.

[chi_sim_layer.zip](https://github.com/tesseract-ocr/tesseract/files/3017840/chi_sim_layer.zip)

**Comment by yixinuestc** [comment]
I used your traineddata, it is very nice,all the result characters are inside my unicharset.

I also comment the "tessedit_load_sublangs chi_sim_vert" in the configure file and retrain again, the traineddata is also okay.

Thank you. I really very appreciate you.
